### PR TITLE
add nonce to comparison

### PR DESCRIPTION
**Pull request by jenr24G**

### CODE REVIEW

File: server/src/processor.rs [hunk 1]
```diff
@@ -138,18 +141,32 @@ impl <'a> SignalProcessor<'a> {
         match signal.signal_status {
             SignalStatus::Settled => match self.current_settled.get_mut(&signal.account) {
                 Some(settled_signal) => {
-                    if signal.height > settled_signal.height {
+                    if signal.nonce > settled_signal.nonce {
```
Comment (mxnkarou) [inline server/src/processor.rs:144]: This is incorrect it should still check against the block height first - if the block heights are the same for both signals you should check against the nonce.

File: server/src/processor.rs [hunk 2]
```diff
@@ -138,18 +141,32 @@ impl <'a> SignalProcessor<'a> {
...
                         *settled_signal = signal.clone();
                     }
                     self.settled_signals.push(signal);
+                    self.invalid_signals.push(settled_signal.clone());
```
Comment (mxnkarou) [inline server/src/processor.rs:148]: Signals need to be reclassified as `Invalid` before being pushed to invalid.

File: server/src/processor.rs [hunk 3]
```diff
@@ -242,3 +226,13 @@ impl <'a> SignalProcessor<'a> {
...
+
+fn is_higher(signal_1: &Signal, signal_2: &Signal) -> bool {
+    signal_1.height > signal_2.height
+        || (
+            signal_1.height == signal_2.height
+                        &&
+            signal_1.nonce > signal_2.nonce
+        )
+
+}
```
Comment (mxnkarou) [inline server/src/processor.rs:238]: Let's make this more descriptive

```
fn is_newer(v1: &Signal, v2: &Signal) -> bool {
    v1.height > v2.height || (v1.height == v2.height && v1.nonce > v2.nonce)
}

Comment (jenr24G) [reply server/src/processor.rs:238]: i figured you'd hate the funky layout. youre worse than a HOA

**Review by mxnkarou — APPROVED** [review]
🥳 🎉

**Pull request merged**